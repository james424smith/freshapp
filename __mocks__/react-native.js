import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
