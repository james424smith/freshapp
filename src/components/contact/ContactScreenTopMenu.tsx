import React from "react";
import { View, TouchableOpacity, ScrollView } from "react-native";
import Text from "../StyledText";
import { ContactType } from "../../interfaces";
import styles from "../../styles/more_screens/ContactsStyles";
import {
  CREW_MANAGER,
  MANNING_AGENT,
  PORT_AGENT,
} from "../../constants/contactConstants";

const categories = [CREW_MANAGER, MANNING_AGENT, PORT_AGENT];

type Props = {
  handleSelection: (type: ContactType) => void;
  selectedType: ContactType;
};

function setStyleOfButton(type: ContactType, selectedType: ContactType) {
  if (type === selectedType) {
    return styles.activeOpacity;
  } else {
    return styles.inactiveOpacity;
  }
}
function setStyleOfText(type: ContactType, selectedType: ContactType) {
  return type === selectedType
    ? styles.activeTouchableOpacityLabel
    : styles.inactiveTouchableOpacityLabel;
}

const ContactScreenTopMenu = (props: Props) => {
  const { handleSelection, selectedType } = props;

  return (
    <ScrollView
      horizontal={true}
      showsHorizontalScrollIndicator={false}
      style={styles.horizontalScrollViewContainer}
    >
      <View style={styles.topContainer}>
        {categories.map((category, i) => (
          <TouchableOpacity
            testID={`category-button`}
            style={[
              styles.buttonStyle,
              setStyleOfButton(category, selectedType),
            ]}
            key={i}
            disabled={category === selectedType}
            onPress={() => {
              handleSelection(category);
            }}
          >
            <Text
              style={[
                styles.textButton,
                setStyleOfText(category, selectedType),
              ]}
            >
              {category}
            </Text>
          </TouchableOpacity>
        ))}
      </View>
    </ScrollView>
  );
};

export default ContactScreenTopMenu;
