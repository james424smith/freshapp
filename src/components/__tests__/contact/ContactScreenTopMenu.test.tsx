import React from "react";
import { render, cleanup, fireEvent } from "@testing-library/react-native";
import ContactScreenTopMenu from "../../contact/ContactScreenTopMenu";

describe("ContactScreenTopMenu snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render AssignmentBasicInformationComponent correctly as per the ui made and selectedType is Manning Agent", () => {
    const temp = jest.fn().mockImplementation((a: string) => {
      a;
    });
    const { toJSON, getAllByTestId } = render(
      <ContactScreenTopMenu
        handleSelection={temp}
        selectedType={"Manning Agent"}
      />
    );

    fireEvent.press(getAllByTestId("category-button")[0]);
    expect(toJSON()).toMatchSnapshot();
    expect(temp).toHaveBeenCalledTimes(1);
  });
  it("should render AssignmentBasicInformationComponent correctly as per the ui made and selectedType is Crew Manager", () => {
    const wrapper = render(
      <ContactScreenTopMenu
        key={0}
        handleSelection={(a: string) => {
          a;
        }}
        selectedType={"Crew Manager"}
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
  it("should render AssignmentBasicInformationComponent correctly as per the ui made and selectedType is Port Agent", () => {
    const wrapper = render(
      <ContactScreenTopMenu
        key={0}
        handleSelection={(a: string) => {
          a;
        }}
        selectedType={"Port Agent"}
      />
    );
    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
