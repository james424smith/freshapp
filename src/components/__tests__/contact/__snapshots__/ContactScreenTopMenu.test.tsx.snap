// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactScreenTopMenu snapshot test should render AssignmentBasicInformationComponent correctly as per the ui made and selectedType is Crew Manager 1`] = `
<RCTScrollView
  horizontal={true}
  showsHorizontalScrollIndicator={false}
  style={
    Object {
      "borderBottomWidth": 0.7,
      "borderColor": "#78849E",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "opacity": 1,
            "padding": 45,
            "textAlign": "center",
          }
        }
        testID="category-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Array [
                Object {
                  "color": "#0093BB",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "justifyContent": "space-around",
                },
                Object {
                  "borderBottomColor": "#0093BB",
                  "borderBottomWidth": 2,
                  "color": "#0093BB",
                  "fontSize": 26.5,
                },
              ],
            ]
          }
        >
          Crew Manager
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "opacity": 0.5,
            "padding": 45,
            "textAlign": "center",
          }
        }
        testID="category-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Array [
                Object {
                  "color": "#0093BB",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "justifyContent": "space-around",
                },
                Object {
                  "color": "#78849E",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "opacity": 0.5,
                },
              ],
            ]
          }
        >
          Manning Agent
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "opacity": 0.5,
            "padding": 45,
            "textAlign": "center",
          }
        }
        testID="category-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Array [
                Object {
                  "color": "#0093BB",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "justifyContent": "space-around",
                },
                Object {
                  "color": "#78849E",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "opacity": 0.5,
                },
              ],
            ]
          }
        >
          Port Agent
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`ContactScreenTopMenu snapshot test should render AssignmentBasicInformationComponent correctly as per the ui made and selectedType is Manning Agent 1`] = `
<RCTScrollView
  horizontal={true}
  showsHorizontalScrollIndicator={false}
  style={
    Object {
      "borderBottomWidth": 0.7,
      "borderColor": "#78849E",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "opacity": 0.5,
            "padding": 45,
            "textAlign": "center",
          }
        }
        testID="category-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Array [
                Object {
                  "color": "#0093BB",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "justifyContent": "space-around",
                },
                Object {
                  "color": "#78849E",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "opacity": 0.5,
                },
              ],
            ]
          }
        >
          Crew Manager
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "opacity": 1,
            "padding": 45,
            "textAlign": "center",
          }
        }
        testID="category-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Array [
                Object {
                  "color": "#0093BB",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "justifyContent": "space-around",
                },
                Object {
                  "borderBottomColor": "#0093BB",
                  "borderBottomWidth": 2,
                  "color": "#0093BB",
                  "fontSize": 26.5,
                },
              ],
            ]
          }
        >
          Manning Agent
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "opacity": 0.5,
            "padding": 45,
            "textAlign": "center",
          }
        }
        testID="category-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Array [
                Object {
                  "color": "#0093BB",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "justifyContent": "space-around",
                },
                Object {
                  "color": "#78849E",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "opacity": 0.5,
                },
              ],
            ]
          }
        >
          Port Agent
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`ContactScreenTopMenu snapshot test should render AssignmentBasicInformationComponent correctly as per the ui made and selectedType is Port Agent 1`] = `
<RCTScrollView
  horizontal={true}
  showsHorizontalScrollIndicator={false}
  style={
    Object {
      "borderBottomWidth": 0.7,
      "borderColor": "#78849E",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "opacity": 0.5,
            "padding": 45,
            "textAlign": "center",
          }
        }
        testID="category-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Array [
                Object {
                  "color": "#0093BB",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "justifyContent": "space-around",
                },
                Object {
                  "color": "#78849E",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "opacity": 0.5,
                },
              ],
            ]
          }
        >
          Crew Manager
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "opacity": 0.5,
            "padding": 45,
            "textAlign": "center",
          }
        }
        testID="category-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Array [
                Object {
                  "color": "#0093BB",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "justifyContent": "space-around",
                },
                Object {
                  "color": "#78849E",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "opacity": 0.5,
                },
              ],
            ]
          }
        >
          Manning Agent
        </Text>
      </View>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "opacity": 1,
            "padding": 45,
            "textAlign": "center",
          }
        }
        testID="category-button"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2A2E43",
                "fontFamily": "Roboto-Regular",
              },
              Array [
                Object {
                  "color": "#0093BB",
                  "fontSize": 26.5,
                  "fontWeight": "600",
                  "justifyContent": "space-around",
                },
                Object {
                  "borderBottomColor": "#0093BB",
                  "borderBottomWidth": 2,
                  "color": "#0093BB",
                  "fontSize": 26.5,
                },
              ],
            ]
          }
        >
          Port Agent
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
